src/rpc/handlers/DepositAuthorized.cpp

```diff
@@ -71,26 +79,81 @@ DepositAuthorizedHandler::process(DepositAuthorizedHandler::Input input, Context
 
     Output response;
 
+    auto it = ripple::SerialIter{dstAccountLedgerObject->data(), dstAccountLedgerObject->size()};
+    auto sleDest = ripple::SLE{it, dstKeylet};
+    bool const reqAuth = ((sleDest.getFieldU32(ripple::sfFlags) & ripple::lsfDepositAuth) != 0u) &&
+        (sourceAccountID != destinationAccountID);
+    bool const credentialsPresent = input.credentials.has_value();
+
+    ripple::STArray authCreds;
+    if (credentialsPresent) {
+        // TODO: move this check into validation.hpp
+        if (input.credentials.value().size() > ripple::maxCredentialsArraySize)
+            return Error{Status{RippledError::rpcINVALID_PARAMS, "an array of CredentialID(hash256)"}};
+
+        for (auto const& elem : input.credentials.value()) {
+            if (!elem.is_string())
+                return Error{Status{RippledError::rpcINVALID_PARAMS, "an array of CredentialID(hash256)"}};
+
+            ripple::uint256 credHash;
+            if (!credHash.parseHex(boost::json::value_to<std::string>(elem)))
+                return Error{Status{RippledError::rpcINVALID_PARAMS, "an array of CredentialID(hash256)"}};
+
+            auto const credKeylet = ripple::keylet::credential(credHash).key;
+            auto const credLedgerObject = sharedPtrBackend_->fetchLedgerObject(credKeylet, lgrInfo.seq, ctx.yield);
+            auto credIt = ripple::SerialIter{credLedgerObject->data(), credLedgerObject->size()};
+            auto sleCred = ripple::SLE{credIt, credKeylet};
+
+            if (!credLedgerObject || (sleCred.getType() != ripple::ltCREDENTIAL) ||
+                ((sleCred.getFieldU32(ripple::sfFlags) & ripple::lsfAccepted) == 0u))
+                return Error{Status{RippledError::rpcBAD_CREDENTIALS}};
+
+            if (checkExpired(sleCred, lgrInfo.closeTime))
+                return Error{Status{RippledError::rpcBAD_CREDENTIALS}};
+
+            if (reqAuth) {
+                auto credential = ripple::STObject::makeInnerObject(ripple::sfCredential);
+                credential.setAccountID(ripple::sfIssuer, sleCred.getAccountID(ripple::sfIssuer));
+                credential.setFieldVL(ripple::sfCredentialType, sleCred.getFieldVL(ripple::sfCredentialType));
+                authCreds.push_back(std::move(credential));
+            }
+        }
+    }
+
+    // If the two accounts are the same OR if that flag is
+    // not set, then the deposit should be fine.
+    bool depositAuthorized = true;
+
+    if (reqAuth) {
+        if (credentialsPresent) {
+            auto const sorted = makeSorted(authCreds);
```

Comment:
```suggestion
            auto const sorted = makeSorted(std::move(authCreds));
```
Probably you will need to update `makeSorted()` as well.

Reply:
Doesn't seem to work with std::move(). Clang-tidy gives `Rvalue reference parameter 'in' is never moved from inside the function body`